ios/ImagePickerManager.m

```diff
@@ -111,40 +108,58 @@ - (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker
     });
 }
 
-- (void)onImageObtained:(NSDictionary<NSString *,id> *)info
-{
-    NSURL *imageURL = [ImagePickerManager getNSURLFromInfo:info];
-    UIImage *image = [ImagePickerManager getUIImageFromInfo:info];
+- (void)picker:(PHPickerViewController *)picker didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)){
+    [picker dismissViewControllerAnimated:YES completion:nil];
+
+    if (results.count == 0) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            self.callback(@[@{@"didCancel": @YES}]);
+        });
+        return;
+    }
+
+    for (PHPickerResult *result in results) {
+        NSItemProvider *provider = result.itemProvider;
+
+        if ([provider hasItemConformingToTypeIdentifier:(NSString *)kUTTypeImage]) {
+            [provider loadDataRepresentationForTypeIdentifier:(NSString *)kUTTypeImage
+                      completionHandler:^(NSData *data, NSError * _Nullable error) {
+                [self onImageObtained:[UIImage imageWithData:data] data:data];
+            }];
+        }
+        else if ([provider hasItemConformingToTypeIdentifier:(NSString *)kUTTypeMovie]) {
+            [provider loadFileRepresentationForTypeIdentifier:(NSString *)kUTTypeMovie
+                                            completionHandler:^(NSURL * _Nullable url, NSError * _Nullable error) {
+                [self onVideoObtained:url];
+            }];
+        }
+    }
+}
 
+- (void)onImageObtained:(UIImage*)image data:(NSData*)data
```

Comment:
Come to think of it, we don't necessarily need to pass the `UIImage` here. We could
```objective-c
UIImage *image = [[UIImage alloc] initWithData:data];
```

Reply:
Fine to put this into a follow-up PR as well imo

Reply:
Yes ideally it should be done as you said. But in UIImagePickerController there is something called edited UIIImage and original UIImage. But the data we obtain is only from original image, you cannot get edited image data. So if the user edited the image then we need to process further using the edited UIImage and use the data from original image to just obtain image type and save it directly for GIF type. I am not sure if I am explaining properly.

This is how previous code has been working so I just used it.

Reply:
I'll look into it. Should be possible to figure out the type and do the editing as you said